core/chaincode/persistence/chaincode_package.go

```diff
@@ -243,7 +243,7 @@ type ChaincodePackageParser struct {
 var (
 	// LabelRegexp is the regular expression controlling
 	// the allowed characters for the package label
-	LabelRegexp = regexp.MustCompile("^[a-zA-Z0-9]+([.+-_][a-zA-Z0-9]+)*$")
+	LabelRegexp = regexp.MustCompile(`^[[:alnum:]][\.\+\-\_[:alnum:]]*$`)
```

Comment:
Escaping everything seems a bit of an overkill, and we don't do it anywhere else. You could just escape the dash (or end with it in the character sequence) and it should achieve the same thing, right?

Reply:
Yes; and there are several other odd regexes that seem incorrect or broken.

Reply:
I actually looked through those, and a cursory look revealed that they are all correct. This is the list of regexp with a dash:
common/flogging/loggerlevels.go:var loggerNameRegexp = regexp.MustCompile(`^[[:alnum:]_#:-]+(\.[[:alnum:]_#:-]+)*$`)
common/viperutil/config_util.go:                var re = regexp.MustCompile(`^(?P<size>[0-9]+)\s*(?i)(?P<unit>(k|m|g))b?$`)
core/ledger/util/couchdb/couchdbutil.go:        re := regexp.MustCompile(`([A-Z])`)
core/container/dockercontroller/dockercontroller.go:    vmRegExp     = regexp.MustCompile("[^a-zA-Z0-9-_.]")
core/container/dockercontroller/dockercontroller.go:    imageRegExp  = regexp.MustCompile("^[a-z0-9]+(([._-][a-z0-9]+)+)?$")
core/scc/lscc/lscc.go:  ChaincodeNameRegExp    = regexp.MustCompile("^[a-zA-Z0-9]+([-_][a-zA-Z0-9]+)*$")
core/scc/lscc/lscc.go:  ChaincodeVersionRegExp = regexp.MustCompile("^[A-Za-z0-9_.+-]+$")
core/chaincode/platforms/node/platform.go:      re := regexp.MustCompile(`^(/)?(src|META-INF)/.*`)
core/chaincode/platforms/golang/platform.go:    re := regexp.MustCompile(`^(src|META-INF)/`)
core/chaincode/platforms/java/platform.go:      filesToMatch := regexp.MustCompile(`^(/)?src/((src|META-INF)/.*|(build\.gradle|settings\.gradle|pom\.xml))`)
core/chaincode/lifecycle/scc.go:        ChaincodeNameRegExp    = regexp.MustCompile("^[a-zA-Z0-9]+([-_][a-zA-Z0-9]+)*$")
core/chaincode/lifecycle/scc.go:        ChaincodeVersionRegExp = regexp.MustCompile("^[A-Za-z0-9_.+-]+$")
core/chaincode/lifecycle/scc.go:        collectionNameRegExp = regexp.MustCompile("^[A-Za-z0-9-]+([A-Za-z0-9_-]+)*$")
core/chaincode/persistence/persistence.go:var packageFileMatcher = regexp.MustCompile("^(.+):([0-9abcdef]+)[.]tar[.]gz$")
internal/configtxlator/integration/cors_test.go:                        address := regexp.MustCompile("127.0.0.1:[0-9]+").FindString(string(errBuffer.Contents()))
internal/ccmetadata/validators.go:      regexp.MustCompile("^META-INF/statedb/couchdb/indexes/.*[.]json"):                                                couchdbIndexFileValidator,
internal/ccmetadata/validators.go:      regexp.MustCompile("^META-INF/statedb/couchdb/collections/" + AllowedCharsCollectionName + "/indexes/.*[.]json"): couchdbIndexFileValidator,

In all cases, the dash is first, last, or right after a character sequence

Reply:
... semantically incorrect. In particular, things like negative version numbers aren’t really expected. I’m not touching any of that since it’s all legacy with more risk than this.

... and you *closed* this agin.

Reply:
ah yes, I only meant that the dash, which was incorrectly set without an escape sequence in the case of LabelRegexp seemed to be correct everywhere else.